.github/workflows/build_phpstanrules_prefixed.yaml

```diff
@@ -0,0 +1,115 @@
+# builds the content of https://github.com/symplify/phpstan-rules
+
+# inspiration from https://github.com/phpstan/phpstan-src/blob/master/.github/workflows/phar.yml
+# and from https://github.com/rectorphp/rector/blob/main/.github/workflows/build_scoped_rector.yaml
+
+name: Build phpstan-rules Prefixed
+
+on:
+    push:
+        branches:
+            - main
+
+        # see https://github.community/t/how-to-run-github-actions-workflow-only-for-new-tags/16075/10?u=tomasvotruba
+        tags:
+            - '*'
+
+env:
+    # see https://github.com/composer/composer/issues/9368#issuecomment-718112361
+    COMPOSER_ROOT_VERSION: "dev-main"
+
+jobs:
+    build_ecs_prefixed:
+        runs-on: ubuntu-latest
+        steps:
+            # make dev build run a bit to avoid collission with tagged build
+            -
+                if: "!startsWith(github.ref, 'refs/tags/')"
+                run: sleep 120
+
+            -
+                uses: actions/checkout@v2
+                # this is required for "WyriHaximus/github-action-get-previous-tag" workflow
+                # see https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches
+                with:
+                    fetch-depth: 0
+
+            -
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: 8.0
+                    coverage: none
+
+            -   run: composer update --no-progress --ansi
+
+            # 1. install package dependencies
+            -   run: packages/monorepo-builder/bin/monorepo-builder localize-composer-paths packages/phpstan-rules/composer.json --ansi
+
+            -
+                run: composer install --working-dir packages/phpstan-rules --ansi --no-dev
+
+            # 2. prepare build directory
+            -   run: cp -r packages/phpstan-rules/. phpstan-rules-build
+            # remove test files
+            -   run: rm -rf phpstan-rules-build/tests phpstan-rules-build/packages/cognitive-complexity/tests phpstan-rules-build/packages/object-calisthenics/tests
+
+            # 3. downgrade
+            -   run: sh packages/phpstan-rules/build/downgrade.sh phpstan-rules-build
+
+            # 4. prefix
+            -   run: sh packages/phpstan-rules/build/build-scoped.sh phpstan-rules-build phpstan-rules-prefixed-downgraded
```

Comment:
to ease further copy/past'ing of this recipe, I have adjusted the filnames to be more generic

`downgrade-ecs.php` -> `downgrade.sh`
`build-scoped.sh` -> `build-ecs-scoped.sh`